### PR TITLE
Deepcopy

### PR DESCRIPTION
**Pull request by registergoofy**
change deepcopy dep lib

### CODE REVIEW

File: pkg/parser/runtime.go [hunk 1]
```diff
@@ -292,11 +293,11 @@ func Parse(ctx UnixParserCtx, xp types.Event, nodes []Node) (types.Event, error)
...
 				isStageOK = true
 				if ParseDump {
-					evtcopy := types.Event{}
-					if err := types.Clone(&event, &evtcopy); err != nil {
-						log.Fatalf("while cloning Event in parser : %s", err)
-					}
-					StageParseCache[stage][node.Name] = evtcopy
+					// if err := types.Clone(&event, &evtcopy); err != nil {
+					// 	log.Fatalf("while cloning Event in parser : %s", err)
+					// }
```
Comment (buixor) [inline pkg/parser/runtime.go:298]: comments

**Review by buixor — APPROVED** [review]
dead code, and what does the new lib brings ?

**Comment by registergoofy** [comment]
> dead code, and what does the new lib brings ?

The code is not dead. It's use by and for heavy debug purpose, in hub-tests. This dependency has no impact in crowdsec as ParseDump is always false.

**Comment by buixor** [comment]
> > dead code, and what does the new lib brings ?
> 
> The code is not dead, but It's used for heavy debug purpose only in hub-tests. This dependency has no impact in crowdsec as ParseDump is always false.

No I mean : what does this new lib brings ?
And for dead code I meant the comments sorry

**Comment by registergoofy** [comment]
> > > dead code, and what does the new lib brings ?
> > 
> > 
> > The code is not dead, but It's used for heavy debug purpose only in hub-tests. This dependency has no impact in crowdsec as ParseDump is always false.
> 
> No I mean : what does this new lib brings ?
> And for dead code I meant the comments sorry

The lib we used until now is not able to clone the entire types.Event struct. Deepcopy can. As this as few to no impact on prod, there's no pain in changing this.

**Pull request merged**